source/fundamentals/serialization/poco.txt

```diff
@@ -203,26 +250,303 @@ The following code sample maps the ``Identifier`` property to the
+   ``[BsonId(IdGenerator)]`` attribute, the driver would generate a non-COMB GUID
+   for the ID.
+
+You can also specify an ``IIdGenerator`` type while registering the class map, as shown in
+the following example:
+
+.. code-block:: csharp
+   :copyable: true
+
+   BsonClassMap.RegisterClassMap<House>(classMap => 
+   {
+      classMap.AutoMap();
+      classMap.MapIdMember(h => h.Id).SetIdGenerator(CombGuidGenerator.Instance);
+   });
+
+.. tip:: Specify an ``IIdGenerator`` for Multiple Classes
+
+   You can use the ``RegisterIdGenerator()`` method to specify a single ``IIdGenerator``
+   for all ``Id`` properties of a certain data type. The following code example instructs
+   the driver to use the ``CombGuidGenerator`` type for all ``Guid`` IDs:
+
+   .. code-block:: csharp
+      :copyable: true
+   
+      BsonSerializer.RegisterIdGenerator(
+        typeof(Guid),
+        CombGuidGenerator.Instance
+      );
+
+The {+driver-short+} also includes ``IIdGenerator`` types that validate the ``Id``
+property and throw an exception if the ID is invalid. The following table lists these
+types:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 10 
+
+   * - ID Validation 
+     - ``IIdGenerator`` Type 
+   * - Not null 
+     - ``NullIdChecker``
+   * - Not all zeroes 
+     - ``ZeroIdChecker<T>``
+
+In the following code example, if the ``Id`` property of the ``House`` class contains
+the default value (``null``), the driver will throw an exception:
+
+.. code-block:: csharp
+   :copyable: false
+   :emphasize-lines: 3
+   
+   public class House
+   {
+       [BsonId(IdGenerator = typeof(NullIdChecker))]
+       public Guid Id { get; set; }
+   }
+
+Customize DateTime Serialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To customize how the {+driver-short+} serializes ``DateTime`` properties, use the
+``[BsonDateTimeOptions()]`` attribute and specify the desired setting as an argument.
+
+If a ``DateTime`` property represents only a date, you can apply the
+``[BsonDateTimeOptions(DateOnly = true)]`` attribute to it.
+If you do, the driver won't perform time-zone conversion on the value.
+
+In the following code example, the ``PatientRecord`` class uses a ``DateTime`` for the
+``DateOfBirth`` property. The ``[BsonDateTimeOptions(DateOnly = true)]``
+attribute indicates that the property contains only a date.
+
+.. code-block:: csharp
+   :copyable: true
+   :emphasize-lines: 5
+
+   public class PatientRecord
+   {
+       public Guid Id { get; set; }
+
+       [BsonDateTimeOptions(DateOnly = true)]
+       public DateTime DateOfBirth { get; set; }
+   }
+
+You can also use the ``[BsonDateTimeOptions()]`` attribute to specify the ``DateTimeKind``
+of a ``DateTime`` property. In the following code example, the ``PatientRecord`` class
+uses a ``DateTime`` for the ``AppointmentTime`` property.
+The ``[BsonDateTimeOptions(Kind = DateTimeKind.Local)]`` attribute indicates
+that the time component of the property's value is in local time. When the
```

Comment:
Q: what does "local" time mean? does the driver pull this from user workspace settings or something?

Reply:
The existing docs don't clarify, but either workspace/computer settings or possibly MongoDB settings